### DISCUSSION TITLE
[Bug] scoop cannot even install git on arm64

### DISCUSSION THREAD

**Discussion by kevin-stuart** [ScoopInstaller/Scoop#5379] [Help and Questions] [answered]
#### Current Behavior
The command `scoop install git` fails to install git.

This is with VMWare Windows 11 ARM64 installation.

#### Expected Behavior
The command `scoop install git` installs some version of git, maybe just a x86/x64 version for Windows.

#### Additional context/output
This is with VMWare Windows 11 ARM64 installation.

Unfortunately, I cannot get the output from VM. However, it seems to have downloaded the 7z file. Then is says "Extrating dl.7z" and then it fails with "Invoke-ExternalCommand" and says that the parameter "FilePath" is NULL. It underlines $7zPath in the error.

Interestingly, 7z works in the command line and seems to be installed and can be called fine.

### System details

**Windows version:** 11

**OS architecture:** arm64

**PowerShell version:** 5.1.22621.963

**Comment by rashil2000** [comment]
Do you have an antivirus running, which might be interfering?

**Reply by kevin-stuart** [reply]
No, not to my knowledge, it is a fresh install of Windows.

**Comment by kevin-stuart** [comment, marked as answer]
I investigated a bit more. I added the entire scoop directory to the exclusions in Windows defender. I also checked, the install package was downloaded to the scoop cache directory. I manually installed git. Afterwords, I could add the extras bucket to scoop. Then I tried to install mpv and basically got the same error I copied away to a file to paste here (some parts of the powershell output seemed to have been localized):

```
Installing 'mpv' (0.35.1) [64bit] from extras bucket
Loading mpv-0.35.1-x86_64.7z from cache
Checking hash of mpv-0.35.1-x86_64.7z ... ok.
Extracting mpv-0.35.1-x86_64.7z ... Invoke-ExternalCommand : Das Argument für den Parameter "FilePath" kann nicht überprüft werden. Das Argument ist NULL
oder leer. Geben Sie ein Argument an, das nicht NULL oder leer ist, und führen Sie den Befehl erneut aus.
In C:\Users\...\scoop\apps\scoop\current\lib\decompress.ps1:45 Zeichen:38
+     $Status = Invoke-ExternalCommand $7zPath $ArgList -LogPath $LogPa ...
+                                      ~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Invoke-ExternalCommand], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Invoke-ExternalCommand

Failed to extract files from C:\Users\...\scoop\apps\mpv\0.35.1\mpv-0.35.1-x86_64.7z.
Log file:
  ~\scoop\apps\mpv\0.35.1\7zip.log

Please try again or create a new issue by using the following link and paste your console output:
https://github.com/ScoopInstaller/Extras/issues/new?title=mpv%400.35.1%3a+decompress+error
```

**Reply by rashil2000** [reply]
What's present in the log file?

**Reply by kevin-stuart** [reply]
To my surprise, there are no log files in scoop\apps\mpv\0.35.1\

**Reply by kevin-stuart** [reply]
Isn't this issue worthy of being an issue instead of a discussion?

**Reply by rashil2000** [reply]
It seems like something is preventing PowerShell scripts from running 7zip. 

An actionable issue hasn't been identified yet; this is why it's a discussion.

**Reply by kevin-stuart** [reply]
I tried looking at all events in the Windows Event Viewer. I found nothing, just as there are no log files. I disabled Microsoft Defender and there is no other security software that should prevent it from running any programs. I can run 7zip without any problems. Since this is a fresh install without having made any modifications to Windows, I assume that this problem should always occur on any Windows 11 installation on arm64. I never had any issues with scoop on x86/x64, so I think it is a problem with scoop on arm64.

If this problem always occurs and is not a local problem, it seems to be bug - or we could call it an issue. Actionability should not mean that something is not an issue; if you already know an action to a problem you may already be close to solving it. But I understand that it may be desirable to segregate issues that you think you can actively work on to solve them.

If I can help in some way and you tell me what to do, I can try helping. From my point, I do not know how I can help out more by myself; I tried to look at all output I could think of to supply you with more information to figure out what is happening. To me, it looks like scoop for some reason behaves differently on arm64. Do you have a suggestion how we can figure out more?

**Reply by niheaven** [reply]
You don't set `use_external_7zip`, right? It is something that scoop could not find 7zip. And do you install 7zip by scoop or by yourself? If latter, please run `scoop config use_external_7zip true` and try again.

**Reply by kevin-stuart** [reply]
I did a `where 7z` and it showed me a path to WindowsApps - I think this is because I installed NanaZip for Windows 11 integration before this time around. Setting `use_external_7zip` worked and I could install `mpv`. Afterwards I used `scoop install 7zip` and it worked too; I did not check if `scoop config use_external_7zip true` is persisted, but in a new terminal, I could install now again after 7zip was installed by scoop now. I have installed NanaZip on another machine too, but did not have problems there.

So the issue for me is somewhat solved, although I still do not completely understand: Why did scoop have problems? Could it not find 7z in the path? But I did in the terminal. Could it not install its own version? What does it usually do by default? Did it not install its own version? I think I might have installed NanaZip because I could not install a program for decompressing with scoop, so I went to get NanaZip to unpack the archive containing mpv by hand.

Anyway, thank you for the suggestion.

**Reply by r15ch13** [reply]
So, after fiddling around for 10 hours to get Windows 11 ARM64 running in a VM, I got the following output.
7z works and could extract git despite the errors.

![Snipaste_2023-02-18_17-00-08](https://user-images.githubusercontent.com/432127/219875995-b17c918a-98a0-408b-8d66-bf90ad84757a.png)
![Snipaste_2023-02-18_17-00-20](https://user-images.githubusercontent.com/432127/219876001-2237eb96-d0f9-4ae4-babe-fb9f6fa00c28.png)
![Snipaste_2023-02-18_17-00-32](https://user-images.githubusercontent.com/432127/219876004-2518b890-f3b8-4571-893d-20d2bbec54e8.png)

**Reply by niheaven** [reply]
![image](https://user-images.githubusercontent.com/5832170/219876995-7eb3c15a-634b-401b-abd0-faaace23662c.png)

Works fine, and the error in `pre_install` is `access denied`? So strange...

The error in `post_install` is file not existed, strange too...

**Reply by niheaven** [reply]
OK @kevin-stuart, the logic of dependencies is broken in your case. Scoop should install 7zip first before install mpv, and we never meet it before.

Will check for it but it's difficult IMO :(

**Reply by r15ch13** [reply]
Ah, the missing .reg files are not the problem. The main bucket is broken because the VM keeps crashing while extracting, so it only has the 7zip.json and git.json files. :smile: 
After adding the `*.reg`-files in, it only shows the access denied error for `7zr.exe`. Probably because Defender is still checking it (the VM is really slow)

![image](https://user-images.githubusercontent.com/432127/219878126-d6793dca-0365-439f-ba7f-0f2e1596794a.png)
